app/domain/authentication/common/authenticate_with_resource_restrictions.rb

```diff
@@ -0,0 +1,50 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    AuthenticateWithResourceRestrictions = CommandClass.new(
+        dependencies: {
+            extract_resource_restrictions:  ExtractResourceRestrictions.new,
+            validate_resource_restrictions: ValidateResourceRestrictions.new,
+            validate_request_parameters:    ValidateParametersMatchResourceRestrictions.new,
```

Comment:
the dependency name should match the class name. I would change the class name to `ValidateRequestParametersMatchResourceRestrictions `